# Getting started with OpenCV

I will be running on Windows, not building from source, with Python. Which for my choices, requies me to input " pip3 install opencv-python ", in the command prompt to install OpenCV. To find your command , go to --> https://opencv.org/get-started/ 

Then to Verify that OpenCV is installed correctly, we can run a simple example. 
Download any image and write down its relative path to the folder.

In [3]:
import cv2 as cv
#my image is in the same folder as this file.
#my file name is nebula.jpg
img = cv.imread("nebula.jpg")

#this displays the image
cv.imshow("Display window", img)
k = cv.waitKey(0) # Wait for a keystroke in the window



The picture I choose seems too big to fit, it looks zoomed in.
I know that my display is of 2560x1440 resolution. I want to know what is the height, width, and the amount of channels that my image has.

In [7]:
# Get original height,width, and channels with shape()
h,w,c = img.shape
print("Original Height and Width:", h,"x", w)
print("Number of channels:", c)

Original Height and Width: 7462 x 7857
Number of channels: 3


Now that we have the images resolution, we can scale it to fit perfectly on our screen (in my case, the screen is 2560 x 1440, and the image I choose is 7462 x 7857). We'll use .resize(),which requires us to create a 2D vector, representing the pixels of the image. The interpolation allows us to specify different methods for resizing the image. Different methods work better for some application than others.

In [12]:
# Set rows and columns 
# lets downsize the image using new  width and height
down_w = 2560
down_h = 1440
down_v = (down_w, down_h)
resized_img = cv.resize(img, down_v, interpolation= cv.INTER_LINEAR)

After resizing the image, we'll redisply the image to verify that it was resized.

In [14]:
#this displays the image
cv.imshow("Resized image", resized_img)
k = cv.waitKey(0) # Wait for a keystroke in the window

We can now view the entire image. And are ready to use OpenCV

## further info: <br>
link to image used: https://unsplash.com/photos/rTZW4f02zY8 <br>
link to cv2.resize() documentation: https://docs.opencv.org/4.x/da/d54/group__imgproc__transform.html#ga47a974309e9102f5f08231edc7e7529d 
